ui/app/src/components/BulkUpload.tsx

```diff
@@ -352,8 +352,8 @@ const DropZone = React.forwardRef((props: DropZoneProps, ref: any) => {
       });
   }
 
-  function checkFileExist(newFile: File) {
-    return !uppy.getFiles().some((file) => file.name === newFile.name && file.type === newFile.type);
+  function checkFileExist(newFile: File & { relativePath?: string }) {
+    return !uppy.getFiles().some((file) => file.meta.relativePath === newFile.relativePath && file.type === newFile.type);
```

Comment:
Why is the relative path in different locations of the file objects? (i.e. file.meta.relativePath vs file.relativePath)

Reply:
The original logic is from here:
```
interval(50)
      .pipe(takeUntil(cancelRequestObservable$), take(files.length))
      .subscribe((index) => {
        const file: File & { relativePath?: string } = files[index];
        checkFileExist(file) &&
          uppy.addFile({
            name: file.name,
            type: file.type,
            data: file,
            meta: {
              relativePath: file.relativePath || null
            }
          });
      });
```

We only call `checkFileExist` function from here.
As you can see, the variable `file` is of `UppyFile` interface with `meta.relativePath`, the `newFile` is of type `File` so they have different location.